src/coi/deltachat_mode.rs

```diff
@@ -0,0 +1,45 @@
+/// COI-related overrides for Deltachat.
+///
+/// In case a COI server with message filter "active" or "seen" is present, the server is
+/// responsible for moving message. In that case DeltaChat should be stopped from moving messages
+/// and the mvbox thread should listen for messages in the `mvbox_folder_override` and the inbox
+/// thread should listen for messages in `inbox_folder_override`.
+pub struct CoiDeltachatMode {
+    pub server_side_move_enabled: bool,
+    pub inbox_folder_override: Option<String>,
+    pub mvbox_folder_override: Option<String>,
+}
```

Comment:
Do we really need to have the separation between COI and DeltaChat be visible outside (i.e. make everything conditional on `server_side_move_enabled`? Just having an `inbox_folder` and an optional `mvbox_folder` would probably remove some `if`s throughout the code base. Only the code writing to "configured_mvbox_folder" might need to be changed to update the mode here.

Reply:
The reason why `server_side_move_enabled` is there is that we have to disable moving messages in `dc_do_heuristics_moves` for instance if COI is active. I don't want to overload the meaning of the "overrides" too much. Overriding the inbox path should not necessarily mean that Coi is enabled. Actually, without the unit tests, we could get rid of `inbox_folder_override` and use "INBOX" all the time.

Reply:
My question was about the dependency between the flag and the folder names, not about the existence of the flag. The idea was instead of having a folder name and an override for it, to just have one variables for the final folder name (one for each thread). So that `dc_is_*` called by `do_heuristics_moves` just compare with the stored string instead of checking the flag and reading the config each time.

Reply:
You mean a configuration for each thread, which folder name to use, unrelated to Coi.
Yes, this would have been a better approach in the first place.

Reply:
Depends on what you mean by "configuration". In DCC this usually means persistent entries in the SQLite database. With "variable" I mean a variable in RAM. Like what the `*_folder_override` variables already are, except without the `_override` suffix, and reading the configured mvbox at most once, if client-side filtering is enabled (as opposed to every time `dc_is_mvbox` is called).

The configuration in the DB would stay the same, i.e. store the folder which is used for client-side filtering.